Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4689,18 +3825,174 @@ - (void)jsqDidChangeLayout
     }
 }
 
-#pragma mark - Class methods
+#pragma mark - View Items
+
+// This is a key method.  It builds or rebuilds the list of
+// cell view models.
+- (void)reloadViewItems
+{
+    NSMutableArray<ConversationViewItem *> *viewItems = [NSMutableArray new];
```

Comment:
> since we call reloadViewItems pretty often

We should be calling `reloadViewItems` pretty rarely, when messages in this conversation change.  

> performBatchUpdates

Of all of this work, the changes around `performBatchUpdates` are least satisfactory so far.  See the next PR which is heavily related to this.